crates/module-system/sov-modules-api/src/state/accessors/internals.rs

```diff
@@ -431,6 +438,20 @@ impl<T> RevertableWriter<T> {
     }
 }
 
+impl<S: Spec> RevertableWriter<StateCheckpoint<S>> {
+    /// Change set resulting from transaction execution.
+    pub fn changes(&self) -> TxChangeSet {
+        TxChangeSet {
```

Comment:
Do we need to handle the uncommitted writes from the `StateCheckpoint`? Or maybe it should be a param that we pass explicitly so that we can make the decision based on the call site (i.e. if we use this method *before* committing the revertable writer for the first time, this is ok; but if we've already committed any other revertable writers it may not be)